corehq/form_processor/backends/sql/processor.py

```diff
@@ -118,8 +124,11 @@ def save_processed_models(cls, processed_forms, cases=None, stock_result=None):
                 # Save deprecated form first to avoid ID conflicts
                 if processed_forms.deprecated:
                     FormAccessorSQL.update_form(processed_forms.deprecated, publish_changes=False)
+                    # in case of rollback, vault entry pk is cleared through tracked models on form
```

Comment:
I don't follow this comment and I also don't understand why there would be vault entries on a deprecated form since that form would have already been processed.

Reply:
From reading this method and code around it, i noticed that other models are stored in `all_models` and then in case of failure their pk is reset later on in case of `DatabaseError`. I didn't add vault_entries in `all_models` because they are added as tracked models on the form for which pk is reset automatically.
Just added the comment to flag that.
`save_processed_models` is called during `SubmissionPost.run`, why do you say that this will be a deprecated form?

Reply:
but it is also called in case of a normal form. if you see 
https://github.com/dimagi/commcare-hq/blob/f3d78aef9343449d99439ebd1e18ff4d3041ec7e/corehq/form_processor/submission_post.py#L319-L338
that method then calls `save_processed_models` on interface.

Reply:
https://github.com/dimagi/commcare-hq/pull/28161/commits/a6d677b20304960fbdedd146c33211d58a8e0bd7
a minor improvement from what you explained. A duplicate form should not be needed so we should just need to worry about submitted form.